lsp/server/src/diagnostic/DiagnosticSettings.ts

```diff
@@ -0,0 +1,62 @@
+    /*
+ * Copyright (c) 2024, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+
+export type DiagnosticId = 
+    | 'missspelled-uiapi' 
+    | 'adapters-local-change-not-aware'; 
```

Comment:
I understand why you have this `type`. But diagnostic producers in their respective `getId()` also returns hard coded string. Ideally, self-declaring ID should be enough. 

With the way the code is written now, as we see it here, we need to register the string because we want to create something like `Set<DiagnosticId>` in line 18 below. And for this reason I prefer something more like
```
export type DiagnosticId = string;
```
I think you can do this in Typescript but need to verify.

Reply:
Yes, I can use export` type DiagnosticId = string;`.  personally I 
like ```type DiagnosticId = 
    | 'missspelled-uiapi' 
    | 'adapters-local-change-not-aware'; 
```. it prevents a ramdam string is returned if forget to reference from the central place.  This pattern is used extensively in lds.  

if you still think **export type DiagnosticId = string;** is better, I can change it.

Reply:
I like to learn why you like your original approach.

Reply:
sure! 
![image](https://github.com/user-attachments/assets/0a7dff76-fab4-4c8b-a176-d052e190852f)
I like the compiling time error check :)

Reply:
I don't think I'm fully getting it.

You get misspelling because you defined it in `ProducerId`. 

If you don't define any thing in `ProducerId`, then, producer ID of the diagnostic rule can be anything and be self declaring.

So I don't see the benefit in the argument that you're making. Except that, you liked to be checked by the guard rail that you self imposed. 

If I missed seeing the benefit I'm all ears!